proxy/src/control/cache.rs

```diff
@@ -27,13 +27,14 @@ pub enum Exists<T> {
     No,
 }
 
-pub enum CacheChange {
+#[derive(Debug, Eq, PartialEq)]
+pub enum CacheChange<'value, K, V: 'value> {
```

Comment:
The main impact of this change is that changed values are now passed as borrowed references, so it's left up to the `on_change` function that receives the `CacheChange` notification to decide whether or not the value should be cloned. 

If, for example, we wanted an `on_change` function that only logs changes to the cache and only needs a reference to log an added or modified value, we wouldn't have to clone the value. If the value ends up being inserted into some other structure as well, then the `on_change` function would be free to clone it if necessary.